repair/repair.cc

```diff
@@ -1950,7 +1950,33 @@ future<> repair_service::do_rebuild_replace_with_repair(locator::token_metadata_
                 rs.get_metrics().replace_total_ranges = nr_ranges_total;
             }).get();
         }
+        std::unordered_map<sstring, std::unordered_set<gms::inet_address>> source_dc_racks;
+        std::unordered_set<gms::inet_address> source_dc_live_nodes;
+        std::unordered_map<sstring, size_t> lost_nodes_per_rack;
+        const auto& topology = tmptr->get_topology();
+        if (!source_dc.empty()) {
+            try {
+                source_dc_racks = topology.get_datacenter_racks().at(source_dc);
```

Comment:
What if user does not user dc and rack at all and uses simple strategy only?

Reply:
hmm, if simple_snitch is used, it will return 

```
    virtual sstring get_rack() const override {
        return "rack1";
    }

    virtual sstring get_datacenter() const override {
        return "datacenter1";
    }
```

by default.

we should always get a rack.

Reply:
For simple strategy I wanted to ignore source_dc since with multiple data centers there is no guarantee that any of the replicas will be in source_dc.